qiskit/visualization/latex.py

```diff
@@ -345,690 +321,303 @@ def _get_beamer_page(self):
-                                "\\cw \\cwx[-" + str(pos_2 - pos_1) + "]"
-                        else:
-                            self._latex[pos_2][column] = \
-                                "\\cw \\cwx[-" + str(pos_2 - pos_1) + "]"
-                    except Exception as e:
-                        raise exceptions.VisualizationError(
-                            'Error during Latex building: %s' % str(e))
-
-                elif op.op._directive:
-                    if self.plot_barriers:
-                        qarglist = op.qargs
-                        indexes = [self._get_qubit_index(x) for x in qarglist]
-                        indexes.sort()
-
-                        first = last = indexes[0]
-                        for index in indexes[1:]:
-                            if index - 1 == last:
-                                last = index
-                            else:
-                                pos = self.img_regs[self.qubit_list[first]]
-                                self._latex[pos][column - 1] += " \\barrier[0em]{" + str(
-                                    last - first) + "}"
-                                self._latex[pos][column] = "\\qw"
-                                first = last = index
-                        pos = self.img_regs[self.qubit_list[first]]
-                        self._latex[pos][column - 1] += " \\barrier[0em]{" + str(
-                            last - first) + "}"
-                        self._latex[pos][column] = "\\qw"
-                else:
-                    raise exceptions.VisualizationError("bad node data")
-
-            # increase the number of columns by the number of columns this layer used
-            column += num_cols_used
+                num_cols_op = 1
+                if isinstance(op.op, Measure):
+                    self._build_measure(op, column)
 
-    def _get_qubit_index(self, qubit):
-        """Get the index number for a quantum bit.
+                elif op.op._directive:  # barrier, snapshot, etc.
+                    self._build_barrier(op, column)
 
-        Args:
-            qubit (tuple): The tuple of the bit of the form
-                (register_name, bit_number)
-        Returns:
-            int: The index in the bit list
-        Raises:
-            VisualizationError: If the bit isn't found
-        """
-        for i, bit in enumerate(self.qubit_list):
-            if qubit == bit:
-                qindex = i
-                break
-        else:
-            raise exceptions.VisualizationError("unable to find bit for operation")
-        return qindex
-
-    def _ffs(self, mask):
-        """Find index of first set bit.
+                else:
+                    gate_text, _ = self._get_gate_ctrl_text(op)
+                    gate_text = self._add_params_to_gate_text(op, gate_text)
+                    gate_text = generate_latex_label(gate_text).replace(" ", "\\,")
```

Comment:
the `replace` can be moved inside `generate_latex_label`.

Reply:
Done in https://github.com/Qiskit/qiskit-terra/pull/5855/commits/2ff9150c1a0e7d6ccf38280ca16f76696d10697e.